src/plugin/traits.rs

```diff
@@ -0,0 +1,58 @@
+use std::ops::Try;
+
+use super::proto::{
+    request,
+    response::{self, PluginResponse},
+};
+
+pub trait PluginInterface {
+    fn methods(&self, _req: request::Methods) -> response::Methods {
```

Comment:
It can't be static because while that's perfectly okay for the current built-in plugins to use a static slice, the main idea of bringing the plugin system in is to use it mainly for external plugins, and these would respond in runtime, so static refs aren't an option.

About request: it doesn't hurt to have it, and this way we will be able to extend the params here to something more useful then `()` in the future.